src/matlab2tikz.m

```diff
@@ -3692,19 +3694,20 @@ case size(cData,2) % specific y-data is given
     % define arrow style
     arrowOptions = opts_print(m2t, arrowOpts, ',');
 
-    % Append the arrow style to the TikZ options themselves.
-    % TODO: Look into replacing this by something more 'local',
-    % (see \pgfplotset{}).
     m2t.content.options = opts_add(m2t.content.options,...
         sprintf('arrow%d/.style', m2t.quiverId), ...
         ['{', arrowOptions, '}']);
-    % - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-    % return the vector field code
-    str = [str, ...
-        sprintf(['\\',name,' [arrow',num2str(m2t.quiverId), '] ', ...
-        'coordinates{(',format,') (',format,')};\n'],...
-        data)];
-    %FIXME: external
+    
+    tabOpts = 'row sep=crcr';
+    [m2t, table] = makeTable(m2t, repmat({''}, size(data,2)), data);
```

Comment:
If you call it as 
```matlab
variables = {'x','y','z','u','v','w'}; % 3D
variables = {'x',y','u','v'}; % for 2D
[m2t, table] = makeTable(m2t, variables, data);
```

there is no need to specify `x y u v` in the output code below as the table will include them for you.